xgi/utils/utilities.py

```diff
@@ -370,28 +370,48 @@ def convert_labels_to_integers(H, label_attribute="label"):
     Because the old IDs are stored in the "label" attribute for both nodes and edges,
     the old "label" values (if they exist) will be overwritten.
     """
-    from ..core import Hypergraph
+    from ..core import DiHypergraph, Hypergraph, SimplicialComplex
 
-    node_dict = dict(zip(H.nodes, range(H.num_nodes)))
-    edge_dict = dict(zip(H.edges, range(H.num_edges)))
-    temp_H = Hypergraph()
-    temp_H._hypergraph = deepcopy(H._hypergraph)
+    node_dict = dict(zip(net.nodes, range(net.num_nodes)))
+    edge_dict = dict(zip(net.edges, range(net.num_edges)))
 
-    temp_H.add_nodes_from((id, deepcopy(H.nodes[n])) for n, id in node_dict.items())
-    temp_H.set_node_attributes(
+    temp_net = net.__class__()
+    temp_net._hypergraph = deepcopy(net._hypergraph)
+
+    temp_net.add_nodes_from((id, deepcopy(net.nodes[n])) for n, id in node_dict.items())
+    temp_net.set_node_attributes(
```

Comment:
can this not just be `net.copy()`?

Reply:
Unfortunately, because we are renaming all the node IDs we can't do that, because `copy()` keeps the same node and edge IDs.